source/common/upstream/ring_hash_lb.cc

```diff
@@ -58,36 +64,118 @@ HostConstSharedPtr RingHashLoadBalancer::Ring::chooseHost(uint64_t h) const {
   }
 }
 
+// TODO(mergeconflict): Determine whether the locality weights we get here are already adjusted
```

Comment:
I think this behavior is missing today, and we need to do the adjustment.

Reply:
Would it be ok to do that in a subsequent PR?

Reply:
Yep.